cyral/resource_cyral_integration_mfa_duo.go

```diff
@@ -0,0 +1,57 @@
+package cyral
+
+import (
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+func resourceIntegrationMFADuo() *schema.Resource {
+	return &schema.Resource{
+		Description: "Manages [integration with Duo MFA](https://cyral.com/docs/mfa/duo).",
+		CreateContext: CreateResource(
+			ConfExtensionIntegrationCreate(duoMFATemplateType),
+			ConfExtensionIntegrationRead(duoMFATemplateType)),
+		ReadContext: ReadResource(ConfExtensionIntegrationRead(duoMFATemplateType)),
+		UpdateContext: UpdateResource(
+			ConfExtensionIntegrationUpdate(duoMFATemplateType),
+			ConfExtensionIntegrationRead(duoMFATemplateType)),
+		DeleteContext: DeleteResource(ConfExtensionIntegrationDelete(duoMFATemplateType)),
+
+		Schema: map[string]*schema.Schema{
+			"id": {
+				Description: "ID of this resource in Cyral environment",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+
+			"name": {
+				Description:  "Integration display name that will be used in the control plane.",
+				Required:     true,
+				Type:         schema.TypeString,
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+			"integration_key": {
+				Description:  "Integration key name obtained from Duo management console.",
+				Required:     true,
+				Type:         schema.TypeString,
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+			"secret_key": {
+				Description:  "Secret key obtained from Duo management console.",
+				Required:     true,
+				Sensitive:    true,
+				Type:         schema.TypeString,
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+			"api_hostname": {
+				Description:  "API hostname obtained from Duo management console.",
+				Required:     true,
+				Type:         schema.TypeString,
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
```

Comment:
Im assuming that all these fields are in fact required by the API.

Reply:
Yep! If you don's specify them, the API returns an error.